lib/dirList.js

```diff
@@ -10,36 +10,84 @@ const dirList = {
    * @param {function(error, entries)} callback
    * note: can't use glob because don't get error on non existing dir
    */
-  list: function (dir, callback) {
+  list: async function (dir, options) {
     const entries = { dirs: [], files: [] }
-    fs.readdir(dir, (err, files) => {
-      if (err) {
-        return callback(err)
-      }
-      if (files.length < 1) {
-        callback(null, entries)
+    const files = await fs.readdir(dir)
+    if (files.length < 1) {
+      return entries
+    }
+
+    await Promise.all(files.map(async filename => {
+      let stats
+      try {
+        stats = await fs.stat(path.join(dir, filename))
+      } catch (error) {
         return
       }
-      let j = 0
-      for (let i = 0; i < files.length; i++) {
-        const filename = files[i]
-        fs.stat(path.join(dir, filename), (err, file) => {
-          if (!err) {
-            if (file.isDirectory()) {
-              entries.dirs.push(filename)
+      const entry = { name: filename, stats }
+      if (stats.isDirectory()) {
+        if (options.extendedFolderInfo) {
+          entry.extendedInfo = await getExtendedInfo(path.join(dir, filename))
+        }
+        entries.dirs.push(entry)
+      } else {
+        entries.files.push(entry)
+      }
+    }))
+
+    async function getExtendedInfo (folderPath) {
+      const depth = folderPath.split(path.sep).length
+      let totalSize = 0
+      let fileCount = 0
+      let totalFileCount = 0
+      let folderCount = 0
+      let totalFolderCount = 0
+      let lastModified = 0
+
+      async function walk (dir) {
+        const files = await fs.readdir(dir)
+        await Promise.all(files
```

Comment:
This is going to become problematic on very large folders.  You should use p-map instead.

Reply:
What is a very large folder?
I tested it with my home folder (91593 files, 19091 folders)
Promise.all: ~1.7s
p-map: ~1.2s

So there is an improvement